indigo_api/models/works.py

```diff
@@ -359,6 +362,55 @@ def as_at_date(self):
         if dates:
             return max(dates)
 
+    def all_provisions(self):
+        # a list of the element ids of the earliest PiT (in whichever language) of a work
+        # does not include Schedules as they don't have commencement dates
+        ids = []
+
+        def add_ids(toc):
+            for e in toc:
+                if e.id:
+                    ids.append(e.id)
+                if e.children and e.component == 'main':
+                    add_ids(e.children)
+
+        first_expression = self.expressions().first()
+        if first_expression:
+            toc = first_expression.table_of_contents()
+            add_ids(toc)
+
+        return ids
+
+    def main_commencement(self):
+        if self.commencements.exists():
+            main = self.commencements.filter(main=True).first()
+            if main:
+                return main
+
+        return None
+
+    def main_commencement_date(self):
+        if self.commencements.exists():
```

Comment:
This goes to the database twice (.exists() and then .first()) -- instead, just use .first() and check for None.

These methods should use `main_commencement()`